src/aggregator/aggregator/follower_flush_mgr.go

```diff
@@ -279,12 +307,39 @@ func (mgr *followerFlushManager) CanLead() bool {
 	return true
 }
 
+// canLeadNotFlushed determines whether the follower can takeover leadership
+// for the window that was not (yet) flushed by leader.
+// This is case is possible when leader encounters a metric at some resolution
+// window for the first time in the shard it owns.
+func (mgr *followerFlushManager) canLeadNotFushed(windowSize time.Duration) bool {
+	earliestPossibleWindowStart := mgr.nowFn().Add(-windowSize)
+	if mgr.openedAt.Before(earliestPossibleWindowStart) {
+		return true
+	}
+
+	return false
+}
+
 func (mgr *followerFlushManager) canLead(
+	flusherType string,
```

Comment:
make this a const enum generated w/ stringer?

Reply:
Good call--could use  http://github.com/m3db/m3/blob/cb2d5f7ad7b426fa48315b2a946d6127cf61774e/src/aggregator/aggregator/list.go#L766-L766